app/com/lucidchart/open/cashy/views/upload/index.scala.html

```diff
@@ -0,0 +1,43 @@
+@import com.lucidchart.open.cashy.models.User
+@import com.lucidchart.open.cashy.views
+@import com.lucidchart.open.cashy.controllers.{routes=>cashyRoutes}
+@import helper._
+
+@(
+  formCreate: Form[_],
+  buckets: Set[String],
+  uploadError: Option[String] = None
+)(
+  implicit
+  request: play.api.mvc.Request[_],
+  userOption: Option[User]
+)
+
+@com.lucidchart.open.cashy.views.html.layouts.main(userOption=userOption) {
+  <div class="row">
+    <h2>Upload Assets</h2>
+    <div class="well">
+      @helper.form(action=cashyRoutes.UploadController.uploadToS3(), ('enctype -> "multipart/form-data")) {
+        <fieldset>
+          @helper.select(formCreate("bucket"), buckets.map(b => b.toString -> b.toString).toSeq, '_label -> "Bucket", '_showConstraints -> false)
+          @helper.inputText(formCreate("assetName"), '_label -> "Asset Name")
+          @helper.inputFile(formCreate("assetFile"), '_label -> "Asset File", 'size -> 10)
+          <button class="btn btn-primary">Upload</button>
+        </fieldset>
+      }
+      @if(formCreate.hasErrors) {
+        @for(error <- formCreate.errors if error.key.isEmpty) {
```

Comment:
In order to verify that the assetName didn't already exist, and also using the value from the form, I had to verify on the entire form.  So the error message now shows up below the form for this particular error.  I check that the key is empty because formerrors do not have a key, and the other errors are displayed by their respective fields by the helper classes.